contracts/test/PremiumCalc.t.sol

```diff
@@ -2,85 +2,75 @@
-        (uint256 total_premium, uint256 principal_premium, uint256 interest_premium) = premiumCalc.getPremiumPayment(address(loan1));  // USDC required for payment on loan
+contract LoanWithPrincipalOwed {
 
-        assertEq(total,         total_premium);
-        assertEq(principal, principal_premium);
-        assertEq(interest,   interest_premium);
+    uint256 public principalOwed;
 
-        assertEq(interest, principal * premiumCalc.premiumFee() / 10_000);
-
-        bob.makeFullPayment(address(loan1));
-
-        uint256 afterBal = IERC20(USDC).balanceOf(address(bob));
-
-        assertEq(beforeBal - afterBal, total);
+    constructor (uint256 _principalOwed) public {
+        principalOwed = _principalOwed;
     }
 
-    function test_late_premium(uint56 _loanAmt, uint256 apr, uint16 index, uint16 numPayments, uint256 lateFee) public {
-        uint256 loanAmt = constrictToRange(_loanAmt, 10_000 * USD, 100 * 1E9 * USD, true);  // $10k to $100b, non zero
-
-        apr     = apr     % 10_000;
-        lateFee = lateFee % 10_000;
-
-        setUpRepayments(loanAmt, apr, index, numPayments);
-
-        mint("USDC",      address(bob),  loanAmt * 1000);  // Mint enough to pay interest
-        bob.approve(USDC, address(loan1), loanAmt * 1000);
-
-        uint256 beforeBal = IERC20(USDC).balanceOf(address(bob));
-
-        hevm.warp(loan1.nextPaymentDue() + 1);  // Payment is late
-
-        (uint256 total,         uint256 principal,         uint256 interest)         = loan1.getFullPayment();                         // USDC required for payment on loan
-        (uint256 total_premium, uint256 principal_premium, uint256 interest_premium) = premiumCalc.getPremiumPayment(address(loan1));  // USDC required for payment on loan
-
-        // Get late fee from regular interest payment
-        (,, uint256 interest_calc) = repaymentCalc.getNextPayment(address(loan1));
-        uint256 interest_late = lateFeeCalc.getLateFee(interest_calc);  // USDC required for payment on loan
+}
 
-        assertEq(total,        total_premium + interest_late);
-        assertEq(principal,                principal_premium);
-        assertEq(interest,  interest_premium + interest_late);
+contract PremiumCalcTest is DSTest {
 
-        assertEq(interest, principal * premiumCalc.premiumFee() / 10_000 + interest_late);
+    function test_calcType() external {
+        assertTrue(new PremiumCalc(999_999).calcType() == uint8(12));
+    }
 
-        bob.makeFullPayment(address(loan1));
+    function test_name() external {
+        assertTrue(new PremiumCalc(999_999).name() == "FLAT");
+    }
 
-        uint256 afterBal = IERC20(USDC).balanceOf(address(bob));
+    function test_lateFee() external {
+        assertEq(new PremiumCalc(999_999).premiumFee(), 999_999);
+    }
 
-        assertEq(beforeBal - afterBal, total);
+    function test_getPremiumPayment() external {
```

Comment:
Lets refactor this to be a fuzzing test

Reply:
Again though, good test

Reply:
We should also use more realistic values for principalOwed (`1 * 10 ** 6 - 10b * 10 ** 6`)